sdk/typescript/README.md

```diff
@@ -1,25 +1,95 @@
+
+All APIs generated from OpenAPI specification are available after initialization either through `gatewayApi.lowLevel` or by calling `innerClient` on corresponding sub-api. See examples in [Low Level APIs section](#low-level-apis). On top of that, this package provides [high-level counterparts](#high-level-apis) for each sub api. 
+
+## Initialization
+
+Calling static `intialize` method from `GatewayApiClient` class will instantiate all low-level APIs together with their high-level equivalents and return them as a single object.
 
 ```typescript
-import "./node-fetch-polyfill" // Polyfill for fetch required if running in node-js
-import { StateApi, StatusApi, TransactionApi, StreamApi } from "@radixdlt/babylon-gateway-api-sdk";
+import { GatewayApiClient } from '@radixdlt/babylon-gateway-api-sdk'
+
+const gatewayApi = GatewayApiClient.initialize({
+  basePath: 'https://kisharnet-gateway.radixdlt.com',
+})
+```
+
+## Low Level APIs
+
+Low level APIs are generated automatically based on OpenAPI spec. You can get a good sense of available methods by looking at [Swagger](https://rcnet.radixdlt.com/swagger/index.html). There are two ways of using each sub api
 
-const transactionApi = new TransactionApi();
+### Using generated APIs through `innerClient`
 
+```typescript
 async function getTransactionStatus(transactionIntentHashHex: string) {
-    let response = await transactionApi.transactionStatus({
-        transactionStatusRequest: {
-            intent_hash_hex: transactionIntentHashHex
-        }
-    });
-    return response.status;
+  let response = await gatewayApi.transaction.innerClient.transactionStatus({
+    transactionStatusRequest: {
+      intent_hash_hex: transactionIntentHashHex,
+    },
+  })
+  return response.status
 }
+console.log(await getTransactionStatus('266cdfe0a28a761909d04761cdbfe33555ee5fdcf1db37fcf71c9a644b53e60b'))
+```
+
+### Using generated APIs through `lowLevel` object
+
+```typescript
+async function getEntityNonFungibleIds(
+  accountAddress: string,
+  vaultAddress: string,
+  nftAddress: string
+) {
+  return gatewayApi.lowLevel.state.entityNonFungibleIdsPage({
+    stateEntityNonFungibleIdsPageRequest: {
+      address: accountAddress,
+      vault_address: vaultAddress,
+      resource_address: nftAddress,
+    },
+  })
+}
+```
+
+## High Level APIs
+
+High Level APIs will grow over time as we start encountering repeating patterns when dealing with low level APIs. They are supposed to help with most commonly performed task and standardize ways of working with Gateway API and Typescript
+
+### State
+
+- `getEntityDetailsVaultAggregated(entities: string | string[])` - detailed information about entities
+
+### Status
+
+- `getCurrent()` - Gateway API version and current ledger state
+- `getNetworkConfiguration()` - network identifier, network name and well-known network addresses
+
+### Transaction
+
+- `getStatus(txID: string)` - transaction status for given transaction id.
```

Comment:
We should probably say `transaction id (the intent hash)` - just to align with the language that the toolkit is using.